app/src/main/java/com/sdp/swiftwallet/presentation/friend/FriendFragment.java

```diff
@@ -0,0 +1,145 @@
+package com.sdp.swiftwallet.presentation.friend;
+
+import static android.content.ContentValues.TAG;
+
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.SetOptions;
+import com.sdp.cryptowalletapp.R;
+import com.sdp.swiftwallet.common.FirebaseUtil;
+import com.sdp.swiftwallet.presentation.friend.fragments.FriendItem;
+import com.sdp.swiftwallet.presentation.friend.fragments.FriendItemFragment;
+import com.sdp.swiftwallet.presentation.friend.fragments.FriendItemRecyclerViewAdapter;
+import com.sdp.swiftwallet.presentation.wallet.fragments.WalletItemFragment;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+
+public class FriendFragment extends Fragment{
+    private FirebaseAuth mAuth;
```

Comment:
I think we should have as little code as possible that refers directly to Firebase inside our Fragments/Activity. Of course sometimes it's absolutely neccessary so we can't always extract an interface but I think in this situation it might be possible. It's not critical but I think it'll be much easier to test if we can use other interfaces that don't depend on Firebase